src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -657,8 +657,19 @@ public bool TryGetValue(string key, out ModelStateEntry value)
             return _innerDictionary.TryGetValue(key, out value);
         }
 
+        /// <summary>
+        /// Returns an enumerator that iterates through the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="ModelStateDictionary.Enumerator"/> structure for the <see cref="ModelStateDictionary"/>.
+        /// </returns>
+        public Enumerator GetEnumerator()
+        {
+            return new Enumerator(_innerDictionary.GetEnumerator());
```

Comment:
>  don't want to expose Dictionary.Enumerator directly as it's leaking an implementation detail

In what way is that enumerator exposed? The wrapper enumerator you're creating has identical semantics and the layer of indirection hides nothing. Regardless, this isn't a worthwhile change and it has no relation to the rest of this PR. Revert changes to this class.